cluster/src/main/java/org/apache/iotdb/cluster/metadata/CMManager.java

```diff
@@ -0,0 +1,233 @@
+        } catch (IOException ex) {
+          // not all cached, pull from remote
+          allSeriesExists = false;
+          break;
+        }
+      }
+      cacheLock.readLock().unlock();
+      if (allSeriesExists) {
+        return measurementSchemas;
+      }
+
+      pullSeriesSchemas(deviceId, measurements);
+
+      // try again
+      boolean allExist = true;
+      cacheLock.readLock().lock();
+      for (int i = 0; i < measurements.length; i++) {
+        try {
+          MeasurementMeta measurementMeta = mRemoteMetaCache.get(deviceId + measurements[i]);
+          measurementSchemas[i] = measurementMeta.getMeasurementSchema();
+        } catch (IOException ex) {
+          allExist = false;
+          break;
+        }
+      }
+      cacheLock.readLock().unlock();
+      if (!allExist) {
+        throw new MetadataException(deviceId + " has some mesurements not found");
+      }
+      return measurementSchemas;
+    }
+  }
+
+  private void pullSeriesSchemas(String deviceId, String[] measurementList)
+    throws MetadataException {
+    List<String> schemasToPull = new ArrayList<>();
+    for (String s : measurementList) {
+      schemasToPull.add(deviceId + IoTDBConstant.PATH_SEPARATOR + s);
+    }
+    List<MeasurementSchema> schemas = metaPuller.pullTimeSeriesSchemas(schemasToPull);
+    for (MeasurementSchema schema : schemas) {
+      cacheMeta(deviceId + IoTDBConstant.PATH_SEPARATOR + schema.getMeasurementId(), new MeasurementMeta(schema));
+    }
+    logger.debug("Pulled {}/{} schemas from remote", schemas.size(), measurementList.length);
+  }
+
+  @Override
+  public void cacheMeta(String seriesPath, MeasurementMeta meta) {
+    cacheLock.writeLock().lock();
+    mRemoteMetaCache.put(seriesPath, meta);
+    cacheLock.writeLock().unlock();
+  }
+
+  @Override
+  public void updateLastCache(String seriesPath, TimeValuePair timeValuePair, boolean highPriorityUpdate, Long latestFlushedTime) {
+    cacheLock.writeLock().lock();
+    try {
+      MeasurementMeta measurementMeta = mRemoteMetaCache.get(seriesPath);
+      measurementMeta.updateCachedLast(timeValuePair, highPriorityUpdate, latestFlushedTime);
+    } catch (IOException e) {
+      // not found
+    } finally {
+      cacheLock.writeLock().unlock();
+    }
+    // maybe local also has the timeseries
+    super.updateLastCache(seriesPath, timeValuePair, highPriorityUpdate, latestFlushedTime);
+  }
+
+  @Override
+  public TimeValuePair getLastCache(String seriesPath) {
+    try {
+      MeasurementMeta measurementMeta = mRemoteMetaCache.get(seriesPath);
+      return measurementMeta.getTimeValuePair();
+    } catch (IOException e) {
+      // do nothing
+    }
+    return super.getLastCache(seriesPath);
+  }
+
+  @Override
+  public MeasurementSchema[] getSeriesSchemasAndReadLockDevice(String deviceId, String[] measurementList, InsertPlan plan) throws MetadataException {
+    //TODO cluster also need to lock device node
```

Comment:
I think this is urgent, or you can extract the lock method separately. Otherwise, an IllegalMonitorStateException will be thrown because the next step will try to unlock but it is not locked.

Reply:
yes, I will fix it

Reply:
yes, I will fix it

Reply:
Yes, I added that just as a temporary solution, but I still hope you consider to do some refactoring, maybe not in this PR, though, since generating exceptions is somehow burdensome.

Reply:
ok，by the way, I use force-push just because some modifications are small, so I think it could be merged with the previous  commit. Anyway, I will avoid do it again.